pycaw/callbacks.py

```diff
@@ -1,3 +1,108 @@
-# TODO
-# implement easy to use callbacks
-# (is already written, but 1 pull at the time)
+from ctypes import pointer
+
+from comtypes import COMObject
+
+from pycaw.api.audiopolicy import (IAudioSessionControl2, IAudioSessionEvents,
+                                   IAudioSessionNotification)
+from pycaw.api.endpointvolume import IAudioEndpointVolumeCallback
+from pycaw.utils import AudioSession
+
+
+class AudioSessionNotification(COMObject):
+    """
+    SESSION CREATED CALLBACK
+
+    """
+    _com_interfaces_ = [IAudioSessionNotification]
+
+    # "OnSessionCreated" is fired by Windows everytime a new session is created
+    def OnSessionCreated(self, NewSession):
```

Comment:
Its a parameter. I have (had) trouble with this mixup between UpperCamelCase and snake_case. The Windows API is in UpperCamelCase and this mixes up the pycaw code.

I thought it would be the best to keep the UpperCamelCase style in the "original" com calls including functions like OnSessionCreated() (i know OnSessionCreated is not really THE "original" and but hear me out 😉 ).
But when a wrapper simplifies the Windows api then i would suggest using snake_case in function names and variables.

I saw the following in pycaw.utils AudioSession()
```
@property
def State(self):
    s = self._ctl.GetState()
    return s
```
for the user it is not quite clear why it is State and not GeState and he might not know that the AudioSession() object is a "non original" Windows pycaw wrapper.
```
@property
def get_state(self):
    s = self._ctl.GetState()
    return s
```
a consistent use of snake_case would make that clear. What do you think?
(i know this turned more into a general discussion but i had been thinking about this for a while now...)

As in the case above i think its also okay to use snake_case since it wont bother the user.

Reply:
Yeah I get the idea and it's not a bad idea, but I feel like it's a slightly different story.
Since my comment on the parameter didn't impact the user actually nor changes the interface.
We can keep that idea you shared for a potential follow up PR I'd say

Reply:
Okay ;) I will open a feature request or something ;)